storage/innobase/include/trx0vtq.h

```diff
@@ -0,0 +1,40 @@
+#ifndef trx0vtq_h
+#define trx0vtq_h
+
+/* Copyright (c) 2016, MariaDB Corporation.
+
+   This program is free software; you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation; version 2 of the License.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program; if not, write to the Free Software
+   Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA */
+
+#include <vtq.h>
+#include "trx0types.h"
+#include "mem0mem.h"
+#include "rem0types.h"
+
+class vtq_query_t
+{
+public:
+	timeval		prev_query;
+	bool		backwards;
+
+	vtq_record_t	result;
+
+	const char * cache_result(mem_heap_t* heap, const rec_t* rec);
+	const char * cache_result(
+		mem_heap_t* heap,
+		const rec_t* rec,
+		const timeval &_ts_query,
+		bool _backwards);
+};
```

Comment:
Please add `/**` comments before `class` and before each member field and member function.
Could the constructors be defined inline?

Reply:
Does inline ctors/dtors are preferable? Why? It slows down compilation and linking. `-flto` can inline it.

Reply:
Sorry, I mistook the member functions for constructors. Only the default constructors and destructors are defined. I see that both `vtq_query_t::cache_result()` are only used (and defined inline) in vers0vtq.cc.

I also see that `trx_t` encapsulates `vtq_query_t`, so we do need a public declaration of the class, so that the `sizeof()` is known. What if the contents of trx0vtq.h was merged to trx0trx.h?

Reply:
https://github.com/tempesta-tech/mariadb/issues/306